packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/utils/tile_overlay.dart

```diff
@@ -0,0 +1,23 @@
+import '../types.dart';
+
+/// Converts an [Iterable] of TileOverlay in a Map of TileOverlayId -> TileOverlay.
+Map<TileOverlayId, TileOverlay> keyTileOverlayId(
+    Iterable<TileOverlay> tileOverlays) {
+  if (tileOverlays == null) {
+    return <TileOverlayId, TileOverlay>{};
+  }
+  return Map<TileOverlayId, TileOverlay>.fromEntries(tileOverlays.map(
+      (TileOverlay tileOverlay) => MapEntry<TileOverlayId, TileOverlay>(
+          tileOverlay.tileOverlayId, tileOverlay)));
+}
+
+/// Converts a Set of TileOverlays into something serializable in JSON.
+List<Map<String, dynamic>> serializeTileOverlaySet(
```

Comment:
If this is used just for the test I'd keep it in the test file.

Reply:
Both methods are used in tile_overlay_updates